### DISCUSSION TITLE
mushr path generation

### DISCUSSION THREAD

**Discussion by twotwobread** [prl-mushr/mushr#83] [Q&A] [answered]
Hi, I'm the University student in Korea. Last year, I studied auto driving using mushr in the University. At that time, I lacked knowledge and learned only shallow knowledge according to that level, so I didn't wonder and didn't learn. But, now I have a question about mushr. 'How does mushr generate a path?'. For generating a path, what algorithm does Mushr use?   ex) A*, RRT
Please answer this question.

**Comment by MarkusSchiffer** [comment, marked as answer]
Hello,
To answer your question briefly, MuSHR is using a variant of A* called ARA* (anytime repairing A*). While we don't implement it ourselves, our MuSHR GP (global planner) node uses a library called [sbpl](http://sbpl.net) which implements it for us. You can see how we use the library in the [C++ file](https://github.com/prl-mushr/mushr_gp/blob/main/src/planner.cpp) within the mushr_gp repository, and feel free to adapt it into your own code as well! [Here](https://www.cs.cmu.edu/~ggordon/mlikhach-ggordon-thrun.ara-tr.pdf) is the paper describing ARA* if you want even more detail. As mentioned in the [autonomous navigation tutorial ](https://mushr.io/tutorials/noetic-navigation/) (note our new version of MuSHR is still in beta and as such is a work in progress) we are also developing a more lightweight version of the global planner, which will likely use vanilla A* to achieve similar results.

I hope this answers your question!

**Reply by twotwobread** [reply]
My curiosity was completely dispelled. Thank you very much.
